components/info_box/ActionBox.tsx

```diff
@@ -0,0 +1,222 @@
+"use client";
+
+import InfoBox from "./InfoBox";
+import { rubik } from "@/utils/Fonts";
+
+import { Form, Button, FormCheck, Row, Col, InputGroup } from "react-bootstrap";
+
+import LocalTeamBox from "./LocalTeamBox";
+import { submitANForm } from "@/utils/AnApiSubmission";
+import FormCheckInput from "react-bootstrap/esm/FormCheckInput";
+import FormCheckLabel from "react-bootstrap/esm/FormCheckLabel";
+import { useRef, useState, useEffect } from "react";
+
+const emailErrorToErrorMessage = (code: EmailErrorCode) => {
+  switch (code) {
+    case "EMAIL_INVALID":
+      return "Please add a valid email address.";
+    case "SERVER_ERROR":
+      return "Something went wrong signing you up. Please try again?";
+    default:
+      return "";
+  }
+};
+
+const checkErrorToErrorMessage = (code: "MISSING_CONSENT" | null) => {
+  switch (code) {
+    case "MISSING_CONSENT":
+      return "Please tick the signup box.";
+    default:
+      return "";
+  }
+};
+
+type FormData = {
+  emailOptIn: boolean;
+  email: string;
+};
+
+const initialFormState: FormData = {
+  emailOptIn: false,
+  email: "",
+};
+
+const ActionBox = ({
+  constituencyData,
+}: {
+  constituencyData: ConstituencyData;
+}) => {
+  const [subscribed, setSubscribed] = useState<string | null | false>(false);
+  const [formState, setFormState] = useState<FormData>(initialFormState);
+  const [emailError, setEmailError] = useState<EmailErrorCode | null>(null);
+  const [checkError, setCheckError] = useState<"MISSING_CONSENT" | null>(null);
+  const formRef = useRef<HTMLFormElement | null>(null);
+
+  useEffect(() => {
+    //string = subscription Date.now()
+    //null = not subscribed on client
+    //false = on server
+    setSubscribed(window.localStorage.getItem("fwd-subscribed"));
+  }, []);
+
+  const submitForm = async () => {
+    if (
+      formState.emailOptIn &&
+      formState.email &&
+      formRef.current &&
+      !formRef.current.email.validity.typeMismatch
+    ) {
+      //TODO set source codes from current url params.
+      const anResponse = await submitANForm(
+        formState.email,
+        "",
+        constituencyData.constituencyIdentifiers,
+        process.env.NEXT_PUBLIC_AN_CONSTITUENCY_FORM || "",
+        [
+          "be counted",
+          "stop the tories",
+          "movement forward",
+          "election reminders",
+          "join",
+        ],
+        "", // source codes,
+      );
+
+      if (anResponse.ok) {
+        window.localStorage.setItem("fwd-subscribed", Date.now().toString());
```

Comment:
This is just going to transition the email form straight into the local team form right?

Might be worth having a middle state like (just stored in `useState`, not local storage) so we can show a 'subscribed successfully' thing? Otherwise some people will probably think it didn't work.

Could just be a `showSubscriptionSuccess` or similar, and it changes the subscribe box to a "thanks for subscribing..." box with an 'X' to close it or a button or whatever, and if/when you close it you get the local team box?

Reply:
Thinking about :arrow_up: , we perhaps want something similar when we redirect you from the homepage postcode lookup to the constituency page. We could pass a URL param `subscribed=true` or whatever and read it on the clientside to also trigger showing the same "thanks for subscribing" box as I mentioned above...

Although on mobile the normal 'subscribe' box may well be out-of-view, so maybe we actually want a little floating banner for the subscribe success?

A Bootstrap toast would probably work well:
https://react-bootstrap.github.io/docs/components/toasts/

Again, not saying we have to do this now / pre-merge :sweat_smile: definitely something we can look at later & in an iterative fashion.

Reply:
So it won't update the page in place currently because it's useRef not useState but yeah we do need some kind of success feedback for the user. 

I figured do that once we have the new layout etc sorted.